### DISCUSSION TITLE
ignoring certain serverless functions from deployment

### DISCUSSION THREAD

**Discussion by sahibjotsaggu** [vercel/community#514] [Help] [answered]
I'm using puppeteer in my Next.js project API which is causing the size of those serverless functions to exceed the limit on deployment. I'm wondering if there's a way to not build those functions on deployment because I don't need to use them in production.

**Comment by swarnava** [comment, marked as answer]
Indeed, the full puppeteer package cannot fit in a Serverless Function. The maximum size is [50MB](https://vercel.com/docs/v2/platform/limits#serverless-function-size)[](https://packagephobia.com/result?p=puppeteer@5.3.1)[](https://packagephobia.com/result?p=puppeteer-core@5.3.1).

 puppeteer: [![install size](https://packagephobia.com/badge?p=puppeteer@5.3.1)](https://packagephobia.com/result?p=puppeteer@5.3.1)

puppeteer-core: [![install size](https://packagephobia.com/badge?p=puppeteer-core@5.3.1)](https://packagephobia.com/result?p=puppeteer-core@5.3.1)

You can consider using `puppeteer-core` instead.

There are also couple of workaround which you can find here: https://github.com/vercel/vercel/issues/4739#issuecomment-926879578

I will also recommend opening a thread for Next.js community: https://github.com/vercel/next.js/discussions

**Reply by sahibjotsaggu** [reply]
Thanks a lot! I ended up moving the unnecessary api files into `api/` (from `pages/api/`).
